Automattic/wp-calypso
# Block Editor Tracking: Track convert to template part and detach blocks from template part actions

#### Changes proposed in this Pull Request

* Track "Template Part" Block -> Toolbar more menu -> "Detach blocks from template part" action
* Track any block -> Toolbar more menu -> "Make template part" action
* Make sure `replaceBlocks` isn't tracked after "Make template part"

**NOTE:** The https://github.com/Automattic/wp-calypso/pull/53598 PR is a prerequisite for this PR.

We are firing both events twice. One with `block_names` property and one without it. We do this to ensure the event is tracked all the time. The event fired with `block_names` property will fail if there are way too many blocks involved. URL length of the request becomes too long and the browser fails to deliver it.


![Screen Shot 2021-07-09 at 12 16 04 PM](https://user-images.githubusercontent.com/5414230/125115696-8b37a100-e0a0-11eb-9489-062a7fc21c85.png)

#### Testing instructions



1.  Follow steps to enable tracking debugging at PCYsg-nrf-p2
2. Open Site Editor
3.  Select a Paragraph block
4.  Click on 3 dots (Options) in the toolbar menu
5.  Click on "Make Template Part"
6.  Confirm `wpcom_block_editor_convert_to_template_part` event is fired twice. One without `block_names` and one with `block_names` property.
7.  Confirm `replaceBlocks` is triggered, but no events are tracked
8. Select the template part you just created
9. Click on 3 dots (Options) in the toolbar menu
10.  Click on "Detach blocks from template part"
11.  Confim `wpcom_block_editor_template_part_detach_blocks` event is fired twice. One without `block_names` and one with `block_names` property.
12. Confirm `replaceBlocks` is triggered, but no events are tracked ([subscriber system](https://github.com/Automattic/wp-calypso/pull/53598) utilized here)



Related to https://github.com/Automattic/wp-calypso/issues/53410
